src/modules/consumer.js

```diff
@@ -164,6 +200,15 @@ class Consumer {
       // receive message, parse it, execute callback, check if should answer, ack/reject message
       const body = parsers.in(msg);
       try {
+        const shouldContinue = await this.hooks.trigger(this, ConsumerHooks.beforeProcessMessageEvent, {
```

Comment:
Can you give an example for a hook that will decide that we don't to reject the message?

It feels like you are treating hooks as interceptors.. while I think of them as observers - so they don't mess with existing control flow.

Reply:
Normally the hook wouldn't interrupt the flow, but I thought it could perhaps be useful for some cases.

Lets say you want your server to automatically reject all messages coming without a specific header, so you can add a hook to check it and reject it before it reaches any subscribed callback.

This will still run any other hooks and the "after process" hook, so things like instrumentation wouldn't break even if a different hook aborts the flow.

Even if the hook doesn't directly abort the flow with it's return value, by nature it still has access the message and all it's properties and can manipulate them as it needs, so it can still affect the flow indirectly.

We can discuss it further

Reply:
As I said early.. I don't see them as hooks, only as observers that don't interrupt the flow. so I wouldn't go this path with giving hook the power to reject messages (and more-over, it's not needed for what we are doing right now with instrumentation).

Up to you.